src/wp-includes/update.php

```diff
@@ -74,12 +74,27 @@ function wp_version_check( $extra_stats = array(), $force_check = false ) {
 	$current->last_checked = time();
 	set_site_transient( 'update_core', $current );
 
-	if ( method_exists( $wpdb, 'db_version' ) ) {
+	// Populate the DB version.
+	if ( method_exists( $wpdb, 'db_server_info' ) ) {
+		$mysql_version = $wpdb->db_server_info();
+	} elseif ( method_exists( $wpdb, 'db_version' ) ) {
 		$mysql_version = preg_replace( '/[^0-9.].*/', '', $wpdb->db_version() );
 	} else {
 		$mysql_version = 'N/A';
 	}
 
+	// Populate the interface DB interface Type.
+	if ( is_resource( $wpdb->dbh ) ) {
+		// Old mysql extension.
+		$mysql_interface = 'mysql';
+	} elseif ( is_object( $wpdb->dbh ) ) {
+		// mysqli, PDO, custom class.
+		$mysql_interface = get_class( $wpdb->dbh );
+	} else {
+		// Unknown sql extension.
+		$mysql_interface = false;
+	}
```

Comment:
It looks like this will also catch SQL Server as being `mysql`, [at least for ProjectNami](https://github.com/ProjectNami/projectnami/blob/master/wp-includes/class-wpdb.php#L1968-L1973), where [sqlsrv_connect](https://www.php.net/manual/en/function.sqlsrv-connect.php) returns a resource. 

The only benefit I can see from this data ends up being detecting Sqlite, as that's in the `get_class()` _usually_. 
I wonder if this wouldn't be better as `$mysql_interface = is_object( $wpdb->dbh ) ? get_class( $wpdb->dbh ) : get_class( $wpdb );`

Reply:
`$wpdb->dbh` is only a resource on sites running PHP 5.6, so the fallback to `get_class( $wpdb )` will only occur on 5.6. You'll get wonky data if a 5.6 site is running a database drop-in such as LudicrousDB, HyperDB, Query Monitor, or W3 Total Cache.

Reply:
`$wpdb->dbh` can be a resource in PHP7+, specifically for non-MySQL dropins, which is my point that the branch is technically incorrect.  
This isn't an issue for MySQL* but is an issue for the entire purpose of including this branch. 

One would have to reference back to `get_resource_type()` to determine what kind of resource it is.